server.go

```diff
@@ -164,6 +173,12 @@ func (a *App) ServeHTTP(w http.ResponseWriter, r *http.Request) {
     // we don't have the key, compute the remote URL
     kvolumes := key2volume(key, a.volumes, replicas)
 
+    // push to leveldb initially as deleted
+    if !a.PutRecord(key, Record{kvolumes, true}) {
```

Comment:
to be completely safe in the case where a soft delete has occurred on this key, you should first check if the `kvolumes` match existing record and if they do not match return a 403